File: cpp/1065/1065.cpp
#include <iostream>

using namespace std;
bool hansu(int n) {
    if (n<100) return 1;
    else if (n<1000) {
        if(((n/100)-((n%100)/10)) == ((((n%100)/10)) - (n%10))) return 1;
        else return 0;
    }
    else return 0;
}
int main(void) {
    int n;
    cin >> n;
    int cnt=0;
    for(int i=1; i<=n; i++) {
        if(hansu(i)) cnt++;
    }
    cout << cnt;
    return 0;
}

File: python/1065.py
n = int(input())

def hansu(num) :
  tmp = str(num)
  sl_num = list(map(int, tmp))
  if len(sl_num) == 1 or len(sl_num) == 2:
    return 1
  elif len(sl_num) == 3 :
    if sl_num[0] + sl_num[2]== sl_num[1]*2 :
      return 1
    return 0
  else :
    return 0

cnt = 0
for i in range(1,n+1) :
  cnt += hansu(i)

print(cnt)